### PR TITLE
This adds the aggregated-metrics-reader ClusterRole which was missing.

### PR DESCRIPTION
**Pull request by jhohertz**
Signed-off-by: Joe Hohertz <joe@viafoura.com>

@olemarkus 
@kennethaasan 

#### What this PR does / why we need it:

This adds the aggregated-metrics-reader ClusterRole which was missing, and seems to be required for k8s 1.8+ per the metrics-server documentation and default deploy manfiests

See: https://github.com/kubernetes-incubator/metrics-server/blob/master/deploy/1.8%2B/aggregated-metrics-reader.yaml

#### Special notes for your reviewer:

I've wrapped it in the rbac.create conditional, not sure if other conditions are appropriate, let me know if you'd like to see something different here.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/metrics-server/templates/aggregated-metrics-reader-cluster-role.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{{- if .Values.rbac.create -}}
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: system:aggregated-metrics-reader
```
Comment (steven-sheehy) [inline stable/metrics-server/templates/aggregated-metrics-reader-cluster-role.yaml]: You need to use the template fullname prefix for name and same templated labels from the other rbacs.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: stable/metrics-server/templates/aggregated-metrics-reader-cluster-role.yaml (hunk 1 above)
Comment (jhohertz) [reply stable/metrics-server/templates/aggregated-metrics-reader-cluster-role.yaml]: Hmmm, I had thought the name was a special handle, but I can't actually find anything to support that.

I'll try testing with something like:
`name: {{ template "metrics-server.fullname" . }}-aggregated-reader` (and the labels) and re-push if all looks to be working okay.

Comment (jhohertz) [reply stable/metrics-server/templates/aggregated-metrics-reader-cluster-role.yaml]: Looks like this works, should address the concerns.

**Comment by olemarkus** [comment]
/assign

This adds only a ClusterRole? I don't see how this will make anything work more than before. At least something else would need to bind to this ClusterRole for this to have any sort of effect.

**Comment by jhohertz** [comment]
@olemarkus I believe the binding is happening via role aggregation, as described here:

https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles

So it is essentially adding itself via the default edit/view/admin cluster roles, and the intrinsic bindings on those.

**Comment by steven-sheehy** [comment]
It's also the exact change that was done in the upstream [PR](https://github.com/kubernetes-incubator/metrics-server/issues/90).

@jhohertz Your lint charts is failing.

**Comment by jhohertz** [comment]
@steven-sheehy I'm not sure what the linter wants...

It seemed to be complaining there was no newline at the end of Charts.yaml before. I didn't remove one, but tried adding one, and now it fails with no real explanation at all.

Going to undo the newline addition and see what happens?

**Comment by olemarkus** [comment]
Interesting.
/lgtm

**Comment by jhohertz** [comment]
Sorry, trying to satisfy the linter, adding the newline to the CI values file results in complaints of blank lines. I've removed it and not going to touch it again, seems to want conflicting things.

**Comment by steven-sheehy** [comment]
/approve
@olemarkus Can you re-apply your lgtm?

**Pull request merged**